diff --git a/config.def.h b/config.def.h
index 7054c06..d977d8a 100644
--- a/config.def.h
+++ b/config.def.h
@@ -56,6 +56,17 @@ static const Layout layouts[] = {
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", normbgcolor, "-nf", normfgcolor, "-sb", selbgcolor, "-sf", selfgcolor, NULL };
 static const char *termcmd[]  = { "st", NULL };
+static const char *brightness_down[] = { "/usr/bin/sudo", "xbacklight", "-ctrl", "nvidia_backlight", "-dec", "20", NULL };
+static const char *brightness_up[] = { "/usr/bin/sudo", "xbacklight", "-ctrl", "nvidia_backlight", "-inc", "20", NULL };
+static const char *kbd_backlight_down[] = { "/usr/bin/sudo", "xbacklight", "-ctrl", "smc::kbd_backlight", "-dec", "20", NULL };
+static const char *kbd_backlight_up[] = { "/usr/bin/sudo", "xbacklight", "-ctrl", "smc::kbd_backlight", "-inc", "20", NULL };
+static const char *audio_previous[] = { "/usr/bin/mocp", "-r", NULL };
+static const char *audio_toggle[] = { "/usr/bin/mocp", "-G", NULL };
+static const char *audio_next[] = { "/usr/bin/mocp", "-f", NULL };
+static const char *volume_toggle[] = { "/usr/bin/amixer", "-q", "set", "Master", "toggle", NULL };
+static const char *volume_dec[] = { "/usr/bin/amixer", "-q", "set", "Master", "10%-", "unmute", NULL };
+static const char *volume_inc[] = { "/usr/bin/amixer", "-q", "set", "Master", "10%+", "unmute", NULL };
+static const char *eject[] = { "/usr/bin/eject", NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
@@ -82,6 +93,19 @@ static Key keys[] = {
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
+	{ 0,         XF86XK_MonBrightnessDown,     spawn,          {.v = brightness_down } },
+	{ 0,         XF86XK_MonBrightnessUp,       spawn,          {.v = brightness_up } },
+	{ 0,         XF86XK_LaunchA,               spawn,          {.v = dmenucmd } },
+	{ 0,         XF86XK_LaunchB,               spawn,          {.v = dmenucmd } },
+	{ 0,         XF86XK_KbdBrightnessDown,     spawn,          {.v = kbd_backlight_down } },
+	{ 0,         XF86XK_KbdBrightnessUp,       spawn,          {.v = kbd_backlight_up } },
+	{ 0,         XF86XK_AudioPrev,             spawn,          {.v = audio_previous } },
+	{ 0,         XF86XK_AudioPlay,             spawn,          {.v = audio_toggle } },
+	{ 0,         XF86XK_AudioNext,             spawn,          {.v = audio_next } },
+	{ 0,         XF86XK_AudioMute,             spawn,          {.v = volume_toggle } },
+	{ 0,         XF86XK_AudioLowerVolume,      spawn,          {.v = volume_dec } },
+	{ 0,         XF86XK_AudioRaiseVolume,      spawn,          {.v = volume_inc } },
+	{ 0,         XF86XK_Eject,                 spawn,          {.v = eject } },
 	TAGKEYS(                        XK_1,                      0)
 	TAGKEYS(                        XK_2,                      1)
 	TAGKEYS(                        XK_3,                      2)
diff --git a/dwm.c b/dwm.c
index 0362114..fc3b740 100644
--- a/dwm.c
+++ b/dwm.c
@@ -36,6 +36,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xproto.h>
 #include <X11/Xutil.h>
+#include <X11/XF86keysym.h>
 #ifdef XINERAMA
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
